apps/site/assets/ts/schedule/components/schedule-finder/ServiceOption.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from "react";
+import { ServiceOptGroup } from "../../../helpers/service";
+import { Service, DayInteger, ServiceWithServiceDate } from "../../../__v3api";
+
+interface Props {
+  service: ServiceWithServiceDate;
+  group: ServiceOptGroup;
+  servicePeriod: string;
+  multipleWeekdays: boolean;
+}
+
+const dayIntegerToString = (day: DayInteger): string =>
+  [
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday"
+  ][day - 1];
+
+const daysAreConsecutive = (
+  days: DayInteger[],
+  areConsecutive: boolean
+): boolean => {
+  if (!areConsecutive) return areConsecutive;
+  const [day, nextDay, ...rest] = days;
+  if (rest.length === 0) return day + 1 === nextDay;
+  return daysAreConsecutive([nextDay, ...rest], day + 1 === nextDay);
+};
+
+export const serviceDays = ({
+  type,
+  valid_days: validDays
+}: Service): string => {
+  if (type === "saturday" || type === "sunday") return "";
+
+  if (validDays.length === 1) return `${dayIntegerToString(validDays[0])}`;
+
+  if (daysAreConsecutive(validDays, true)) {
+    return `${dayIntegerToString(validDays[0])} - ${dayIntegerToString(
+      validDays[validDays.length - 1]
+    )}`;
+  }
+
+  return `${validDays.map(dayIntegerToString).join(", ")}`;
+};
+
+const ServiceOption = ({
+  service,
+  group,
+  servicePeriod,
+  multipleWeekdays
+}: Props): ReactElement<HTMLElement> => {
+  const isMultipleWeekday =
+    multipleWeekdays &&
+    service.type === "weekday" &&
+    service.typicality !== "holiday_service";
+  return (
+    <option value={service.id}>
+      {isMultipleWeekday
+        ? `${serviceDays(service)} schedule`
+        : service.description}
+      {group !== "holiday" ? ", " : " "}
```

Comment:
<img width="763" alt="image" src="https://user-images.githubusercontent.com/2136286/70349773-69769800-1833-11ea-91e0-c7c51d2d1c71.png">

What happened here? I'm wondering if there's an inconsistency between the `group` value of "holiday" and the `service.typicality` value of "holiday_service" -- the output suggests that in this particular example for the day after thanksgiving, `isMultipleWeekday` is true, `serviceDays(service)` outputs nothing, and line 65 outputs `" "`. But it shows under the Holiday Schedules optgroup, but not rendering with `HolidayServiceOption`.

Reply:
@thecristen For this branch I'm seeing an output that looks very different from what you screenshotted here. For example I see: "Thanksgiving Day, Nov 28" rather than the "Holiday (Sunday schedule) Thanksgiving Day" in your screenshot.

Am I missing something?

I do see "schedule Day after Thanksgiving Day" come through as well though for me it's in future schedules like so:
![image](https://user-images.githubusercontent.com/5806643/70366838-7c578f80-1868-11ea-8d36-881f0badb7fa.png)

Reply:
@ingridpierre The screenshot is what I saw/still see on dev green: https://green.dev.mbtace.com/schedules/751/line 🤷‍♀

Reply:
@ingridpierre @thecristen ah, so I see I left out one important thing, when I said to try this against dev-green, I meant the dev-green _API_. That is, I ran the server locally, but with the env var `V3_URL=https://green.dev.api.mbtace.com`.

When I do that, this is the selector I see for that route:

<img width="648" alt="Screen Shot 2019-12-09 at 1 54 32 PM" src="https://user-images.githubusercontent.com/9827/70463854-b6ed4200-1a8b-11ea-86af-3eceb63e4116.png">

Reply:
Ohhhh, thanks that makes a huge difference. :sweat_smile: Now I can see what you see for the SL4 as well. 👍 

What I get for the 39 bus is pretty different, (for example Monday - Friday instead of "Weekday") -- is this discrepancy expected?

<img width="796" alt="image" src="https://user-images.githubusercontent.com/2136286/70486094-a2766d00-1abe-11ea-9cf8-edd6e24cf902.png">

Reply:
We want it to say "Weekday". Ideally like this ([see my slack thread](https://mbtace.slack.com/archives/C8THJ1NJY/p1575681138010800))
<img width="645" alt="Pasted Graphic 13" src="https://user-images.githubusercontent.com/5806643/70542368-9cbc6e00-1b36-11ea-9269-7cf2ff24569b.png">

so that it ultimately looks like this:
<img width="741" alt="Pasted Graphic 14" src="https://user-images.githubusercontent.com/5806643/70542434-b493f200-1b36-11ea-9b49-f5d1669e4ae2.png">

Reply:
@ingridpierre I just realized that one thing we don't have a card for, is to make the current schedules consistently read "Weekday", "Saturday", or "Sunday" as appropriate, i.e. none of this "Monday-Friday schedule" business anymore. Is that something we need for Thursday or can we live with the current rendering a little longer (assuming that we get rid of redundant schedules)?

Reply:
I found it concerning that the output isn't consistent here between Monday-Friday and Weekday, but let's address that in a followup PR.